boto3/dynamodb/types.py

```diff
@@ -0,0 +1,287 @@
+            return True
+        return False
+
+    def _is_number(self, value):
+        if isinstance(value, (six.integer_types, Decimal)):
+            return True
+        elif isinstance(value, float):
+            raise TypeError(
+                'Float types are not supported. Use Decimal types instead.')
+        return False
+
+    def _is_string(self, value):
+        if isinstance(value, six.string_types):
+            return True
+        return False
+
+    def _is_binary(self, value):
+        if isinstance(value, Binary):
+            return True
+        elif isinstance(value, bytearray):
+            return True
+        elif six.PY3 and isinstance(value, six.binary_type):
+            return True
+        return False
+
+    def _is_set(self, value):
+        if isinstance(value, (set, frozenset)):
+            return True
+        return False
+
+    def _is_type_set(self, value, type_validator):
+        if self._is_set(value):
+            if False not in map(type_validator, value):
+                return True
+        return False
+
+    def _is_map(self, value):
+        if isinstance(value, Mapping):
+            return True
+        return False
+
+    def _is_list(self, value):
+        if isinstance(value, list):
+            return True
+        return False
+
+    def _serialize_null(self, value):
+        return True
+
+    def _serialize_bool(self, value):
+        return value
+
+    def _serialize_n(self, value):
+        number = str(DYNAMODB_CONTEXT.create_decimal(value))
+        if number in ['Infinity', 'NaN']:
+            raise TypeError('Infinity and NaN not supported')
+        return number
+
+    def _serialize_s(self, value):
+        return value
+
+    def _serialize_b(self, value):
+        if isinstance(value, Binary):
+            value = value.value
+        return value
+
+    def _serialize_ss(self, value):
+        return [self._serialize_s(s) for s in value]
+
+    def _serialize_ns(self, value):
+        return [self._serialize_n(n) for n in value]
+
+    def _serialize_bs(self, value):
+        return [self._serialize_b(b) for b in value]
+
+    def _serialize_l(self, value):
+        return [self.serialize(v) for v in value]
+
+    def _serialize_m(self, value):
+        return dict([(k, self.serialize(v)) for k, v in value.items()])
+
+
+class TypeDeserializer(object):
+    """This class deserializes DynamoDB types to Python types."""
+    def deserialize(self, value):
+        """The method to deserialize the DynamoDB data types.
+
+        :param value: A DynamoDB value to be deserialized to a pythonic value.
+            Here are the various conversions:
+
+            DynamoDB                                Python
+            --------                                ------
+            {'NULL': True}                          None
+            {'BOOL': True/False}                    True/False
+            {'N': str(value)}                       Decimal(str(value))
```

Comment:
Good point. I originally was going to check if the number string had a ``.`` in it to determine if it is a decimal or integer, but it is perfectly valid that an integer could be originally passed in as a ``Decimal``.

Given that, I think updating the docs to note it would be best.